# Fusion des datasets annuels et cleaning

Années à fusionner :
- 2017
- 2018
- 2019
- 2020

## Import des librairies

In [1]:
import pandas as pd
import numpy as np
import re
from scipy import stats
from sklearn.feature_extraction.text import CountVectorizer

## Lecture et concaténation des datasets annuels

In [2]:
df1 = pd.read_csv("./Data/dataset-2017.csv")
df2 = pd.read_csv("./Data/dataset-2018.csv")
df3 = pd.read_csv("./Data/dataset-2019.csv")
df4 = pd.read_csv("./Data/dataset-2020.csv")

In [3]:
df = pd.concat([df1,df2,df3,df4])

In [4]:
df.head(20)

code_mission type_mission  exercice code_client  total_assistant  \
0   AAGS2017010396900          AGS    2017.0        5782              2.0   
1   AAGS2017010397000          AGS    2017.0        6565              1.0   
2   AAGS2017010534300          AGS    2017.0        6924              2.0   
3   ACAB2017009616200          CAB    2017.0        9230             12.0   
4   ACAB2017009616300          CAB    2017.0        9000             98.0   
5   ACAB2017009616400          CAB    2017.0        9010             16.0   
6   ACAB2017009616500          CAB    2017.0        9020             58.0   
7   ACAB2017009616600          CAB    2017.0        9050             18.0   
8   ACAB2017009616700          CAB    2017.0        9060              3.0   
9   ACAB2017009616800          CAB    2017.0        9070            101.0   
10  ACAB2017009616900          CAB    2017.0        9090            113.0   
11  ACAB2017009617000          CAB    2017.0        9210             44.0   
12  ACAB2017009617100          CAB    2017.0        9220             56.0   
13  ACAB2017009617200          CAB    2017.0        9240            149.0   
14  ACAB2017009617300          CAB    2017.0        9250             86.0   
15  ACAB2017009617400          CAB    2017.0        9500             95.0   
16  ACAB2017009617500          CAB    2017.0        9700             42.0   
17  ACAB2017009617600          CAB    2017.0        9710             23.0   
18  ACAB2017009627000          CAB    2017.0        8961              9.0   
19  ACAB2017009628300          CAB    2017.0        8937              3.0   

           manager                                       code_article  \
0              JOP                                            ['029']   
1              FRL                                            ['001']   
2              JOP                              ['029', '024', '001']   
3   NON FACTURABLE  ['081', '096', '091', '079', '078', '082', '07...   
4   NON FACTURABLE  ['083', '082', '079', '096', '075', '010', '09...   
5   NON FACTURABLE  ['086', '088', '082', '076', '072', '098', '089']   
6   NON FACTURABLE  ['092', '083', '089', '096', '082', '079', '08...   
7   NON FACTURABLE  ['084', '077', '061', '079', '076', '089', '08...   
8   NON FACTURABLE                                            ['085']   
9   NON FACTURABLE  ['088', '089', '077', '082', '079', '075', '08...   
10  NON FACTURABLE  ['089', '079', '076', '090', '083', '082', '07...   
11  NON FACTURABLE  ['072', '096', '074', '082', '073', '049', '09...   
12  NON FACTURABLE  ['076', '078', '089', '083', '070', '079', '07...   
13  NON FACTURABLE  ['079', '0792', '088', '076', '078', '094', '1...   
14  NON FACTURABLE  ['077', '099', '088', '082', '084', '078', '07...   
15  NON FACTURABLE  ['079', '080', '105', '0792', '075', '0795', '...   
16  NON FACTURABLE  ['075', '070', '097', '088', '074', '082', '07...   
17  NON FACTURABLE               ['070', '072', '075', '0796', '079']   
18             CIT  ['078', '077', '081', '076', '079', '088', '082']   
19             CIT                       ['078', '079', '089', '081']   

                                      libelle_article     temps  \
0                        ['AUTRES TRAVAUX EN SOCIAL']     43.00   
1                           ['RENSEIGNEMENTS CLIENT']      1.00   
2   ['AUTRES TRAVAUX EN SOCIAL', 'FORMALITES DE LI...     31.25   
3   ['COMPTABILITE GENERALE CABINET', 'DOCUMENTATI...   2049.50   
4   ['NON FACTURABLE SOCIAL', 'NON FACTURABLE COMP...   2279.64   
5   ['VERIFICATION FINALE TRAVAUX', 'ARCHIVAGE DOS...    509.80   
6   ['FORMATION PROFESSIONNELLE INTERNE', 'NON FAC...   4958.35   
7   ['NON FACTURABLE JURIDIQUE', 'SECRETARIAT CABI...    726.25   
8                     ['NON FACTURABLE COMMISSARIAT']     20.75   
9   ['ARCHIVAGE DOSSIERS CLASSEMENT', 'ASSISTANCE ...   1516.40   
10  ['ASSISTANCE INFORMATIQUE INTERNE', 'ORGANISAT...   3008.17   
11  ['RENDEZ VOUS PROSPECT', 'DOCUMENTATION', 'REU...

In [118]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35583 entries, 0 to 35582
Columns: 197 entries, code_mission to snc
dtypes: float64(4), int64(183), object(10)
memory usage: 53.5+ MB


### Le dataset final se compose de 35 583 lignes et 197 colonnes

## Vérification des valeurs nulles

In [6]:
df.isna().sum()

code_mission          0
type_mission          0
exercice              0
code_client           0
total_assistant       0
manager               0
code_article          0
libelle_article       0
temps                 0
valorisation_temps    0
site                  0
facturation           0
entrée_clt            0
sortie_clt            0
secteur               0
forme                 0
dtype: int64

In [7]:
df.isnull().sum()

code_mission          0
type_mission          0
exercice              0
code_client           0
total_assistant       0
manager               0
code_article          0
libelle_article       0
temps                 0
valorisation_temps    0
site                  0
facturation           0
entrée_clt            0
sortie_clt            0
secteur               0
forme                 0
dtype: int64

## Données catégorielles

In [8]:
df.dtypes

code_mission           object
type_mission           object
exercice              float64
code_client            object
total_assistant       float64
manager                object
code_article           object
libelle_article        object
temps                 float64
valorisation_temps    float64
site                   object
facturation           float64
entrée_clt              int64
sortie_clt              int64
secteur                object
forme                  object
dtype: object

Colonnes catégorielles :
- 'code_mission'
- 'type_mission'
- 'code_client'
- 'manager'
- 'code_article'
- 'libelle_article'
- 'site'
- 'secteur'
- 'forme'

In [9]:
#changer 'exercice' en catégorielle

In [10]:
#Le code mission est bien un identifiant unique. Il y a 35 740 missions réalisées depuis 2017 à 2020.
df.code_mission.value_counts().sort_values(ascending=True)

ASOP2019012303200    1
AGES2018011613000    1
APOL2018011475000    1
ASOP2017010291600    1
ACOM2017012086800    1
                    ..
AJUR2018008934500    1
ACOM2018010563200    1
ASOP2018011439200    1
ASOP2019012196200    1
ACOM2019009954200    1
Name: code_mission, Length: 35739, dtype: int64

In [11]:
df.type_mission.value_counts()

COM    14580
JUR     7186
SOP     4924
GEJ     2947
GS0     1282
SAP      902
GES      888
GE0      843
GS3      500
GS1      467
POL      327
GEP      293
CAC      219
CAB      141
GE1       53
GJ0       49
GS2       29
GS4       26
TB        25
GJ1       22
AGS       10
GE2        7
GEC        3
CAT        3
GE3        3
GJ2        3
GJ3        2
GJ5        1
GJ4        1
GJ6        1
CAP        1
GJ7        1
Name: type_mission, dtype: int64

- Service Commissaire Aux Comptes : 
    CAC
    CAP
    CAT
- Service Juridique :
    JUR
    GEJ
- Service Paie :
    GEP
    POL
    SOP
- Service social :
    AGS
    GES
    GS0
    GS1
    GS2
    GS3
    GS4

In [12]:
df.code_client.value_counts()

77254    29
4794     28
7149     27
3295     26
7402     26
         ..
A3204     1
A2324     1
A3535     1
A2516     1
A2558     1
Name: code_client, Length: 6403, dtype: int64

In [13]:
df.manager.value_counts()

SLE                      5502
DAC                      4504
GSO                      4103
VAL                      3479
FRL                      3436
EMG                      3370
NAB                      2979
JOP                      2792
THB                      1706
CBA                      1299
STL                      1212
ALD                       377
JOD                       292
CIT                       259
_OLP NE PLUS UTILISER     182
CHB                       119
NON FACTURABLE             80
_FAY NE PLUS UTILISER      42
DIVERS                      5
CAV                         1
Name: manager, dtype: int64

### 19 managers associés dans ce dataset

In [14]:
#Il y a 11 394 listes d'activité sur l'ensemble des missions
df.libelle_article.value_counts().head()

['SECRETARIAT JURIDIQUE ANNUEL']                        3286
['DIVERS JURIDIQUE']                                    1442
['CONTRATS DE TRAVAIL']                                 1262
['AUTRES DECLARATIONS FISCALES']                         931
['DIVERS JURIDIQUE', 'SECRETARIAT JURIDIQUE ANNUEL']     865
Name: libelle_article, dtype: int64

In [15]:
#Quand le site = CLIENT PARTI, cela signifie que le client a quitté le cabinet avant 2017
df[df['site'] == "CLIENT PARTI"]

code_mission type_mission  exercice code_client  total_assistant  \
193   ACOM2017008083600          COM    2017.0       79402              1.0   
791   ACOM2017008292000          COM    2017.0        5711              1.0   
841   ACOM2017008299400          COM    2017.0        5985              1.0   
930   ACOM2017008310700          COM    2017.0        2395              1.0   
1104  ACOM2017008333700          COM    2017.0        6834              1.0   
...                 ...          ...       ...         ...              ...   
5828  AJUR2019010696700          JUR    2019.0        7340              1.0   
7356  AJUR2019011794600          JUR    2019.0        6448              3.0   
8250  ASOP2019012251400          SOP    2019.0        7340              1.0   
4399  AGEJ2020013450900          GEJ    2020.0        6448              1.0   
6125  AJUR2020011020800          JUR    2020.0       A0259              1.0   

     manager                                 code_article  \
193      GSO                                      ['016']   
791      GSO                                      ['060']   
841      DAC                                      ['060']   
930      FRL                                      ['021']   
1104     FRL                                      ['020']   
...      ...                                          ...   
5828     EMG                                      ['060']   
7356     SLE                           ['069', '060', 69]   
8250     EMG  ['F0217', 'F02171', 'F02081', '020', '029']   
4399     SLE                                      ['069']   
6125     EMG                                      ['060']   

                                        libelle_article  temps  \
193                    ['AUTRES DECLARATIONS FISCALES']   0.25   
791                    ['SECRETARIAT JURIDIQUE ANNUEL']   0.50   
841                    ['SECRETARIAT JURIDIQUE ANNUEL']   0.25   
930        ['DECLARATIONS MENSUELLES / TRIMESTRIELLES']   0.50   
1104                                  ['FICHE DE PAIE']   0.50   
...                                                 ...    ...   
5828                   ['SECRETARIAT JURIDIQUE ANNUEL']   0.50   
7356  ['DIVERS JURIDIQUE', 'SECRETARIAT JURIDIQUE AN...   3.25   
8250  ['FORFAIT PRELEVEMENT A LA SOURCE ENTREPRISE',...   4.00   
4399                               ['DIVERS JURIDIQUE']   3.00   
6125                   ['SECRETARIAT JURIDIQUE ANNUEL']   0.50   

      valorisation_temps          site  facturation  entrée_clt  sortie_clt  \
193                16.50  CLIENT PARTI       7160.0           0           0   
791                34.00  CLIENT PARTI          0.0           0           0   
841                18.50  CLIENT PARTI          0.0           0           0   
930                28.00  CLIENT PARTI          0.0           0           0   
1104               28.00  CLIENT PARTI          0.0           0           0   
...                  ...           ...          ...         ...         ...   
5828               31.50  CLIENT PARTI          0.0           0           0   
7356              203.25  CLIENT PARTI        180.0           0           0   
8250              226.00  CLIENT PARTI        170.0           0           0   
4399              229.53  CLIENT PARTI         75.0           0           0   
6125               31.50  CLIENT PARTI        360.0           0           0   

                                                secteur forme  
193                                 Services funéraires  SARL  
791   Location de terrains et d'autres biens immobil...   SCI  
841                                              Divers   SCI  
930   Autres commerces de détail sur éventaires et m...   IND  
1104                                             Divers   IND  
...                                                 ...   ...  
5828                                           Coiffure  SARL  
7356  Location de terrains et d'autres biens immobil...   SCI  
8250       

In [16]:
df.site.value_counts()

GIEN                    6249
OLIVET                  6050
PITHIVIERS              4244
ORLEANS                 3181
VILLEMANDEUR            2766
BRIARE                  2401
MEUNG SUR LOIRE         2284
FONTAINEBLEAU           2187
LA FERTE ST AUBIN       2119
YERRES                  1619
ANGERVILLE              1059
EVRY                     814
CLIENT PARTI             226
CAC                      224
VILLENEUVE LA GUYARD     127
NON FACTURABLE            86
DOSSIERS INTERNES         58
MORSANG SUR ORGE          38
MALESHERBES                5
DIVERS                     2
Name: site, dtype: int64

In [17]:
#Le nombre de missions augmente chaque année
df.exercice.value_counts()

2020.0    9926
2018.0    9019
2019.0    8875
2017.0    7919
Name: exercice, dtype: int64

## Convertir la colonne "exercice" en catégorielle

In [18]:
df.exercice = df.exercice.astype(int).astype(str)

In [19]:
df.dtypes

code_mission           object
type_mission           object
exercice               object
code_client            object
total_assistant       float64
manager                object
code_article           object
libelle_article        object
temps                 float64
valorisation_temps    float64
site                   object
facturation           float64
entrée_clt              int64
sortie_clt              int64
secteur                object
forme                  object
dtype: object

In [20]:
df.head()

code_mission type_mission exercice code_client  total_assistant  \
0  AAGS2017010396900          AGS     2017        5782              2.0   
1  AAGS2017010397000          AGS     2017        6565              1.0   
2  AAGS2017010534300          AGS     2017        6924              2.0   
3  ACAB2017009616200          CAB     2017        9230             12.0   
4  ACAB2017009616300          CAB     2017        9000             98.0   

          manager                                       code_article  \
0             JOP                                            ['029']   
1             FRL                                            ['001']   
2             JOP                              ['029', '024', '001']   
3  NON FACTURABLE  ['081', '096', '091', '079', '078', '082', '07...   
4  NON FACTURABLE  ['083', '082', '079', '096', '075', '010', '09...   

                                     libelle_article    temps  \
0                       ['AUTRES TRAVAUX EN SOCIAL']    43.00   
1                          ['RENSEIGNEMENTS CLIENT']     1.00   
2  ['AUTRES TRAVAUX EN SOCIAL', 'FORMALITES DE LI...    31.25   
3  ['COMPTABILITE GENERALE CABINET', 'DOCUMENTATI...  2049.50   
4  ['NON FACTURABLE SOCIAL', 'NON FACTURABLE COMP...  2279.64   

   valorisation_temps            site  facturation  entrée_clt  sortie_clt  \
0             5231.25      PITHIVIERS       4550.0           0           1   
1               68.00            GIEN        750.0           0           0   
2             3868.00      PITHIVIERS       4804.2           0           0   
3           222874.75  NON FACTURABLE          0.0           0           0   
4           151143.13  NON FACTURABLE          0.0           0           0   

                                             secteur        forme  
0  Fabrication de charpentes et d'autres menuiseries          SAS  
1                           Activités d'architecture         EURL  
2     Fabrication de meubles de bureau et de magasin          SAS  
3                                             Divers  Particulier  
4                                             Divers  Particulier

In [21]:
#Supprimer les activités liées au cabinet comptable, inutile au ML

In [22]:
list_index_cab = df[df['type_mission'] == "CAB"].index
len(list_index_cab)

141

In [23]:
df.drop(list_index_cab, axis=0, inplace=True)

## Bag of Words sur les types de missions

In [24]:
## Création d'un "bag of words" pour analyser les données des types de missions

In [25]:
text_data = np.array(df['type_mission'].values)
text_data

array(['AGS', 'AGS', 'COM', ..., 'TB', 'TB', 'TB'], dtype=object)

In [26]:
# Create the bag of words feature matrix
count = CountVectorizer()
bow = count.fit_transform(text_data)

# Show feature matrix
bow.toarray()

array([[1, 0, 0, ..., 0, 0, 0],
       [1, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       ...,
       [0, 0, 0, ..., 0, 0, 1],
       [0, 0, 0, ..., 0, 0, 1],
       [0, 0, 0, ..., 0, 0, 1]], dtype=int64)

In [27]:
feature_names = count.get_feature_names()

In [28]:
df_types = pd.DataFrame(bow.toarray(), columns=feature_names)
df_types

ags  cac  cap  cat  com  ge0  ge1  ge2  ge3  gec  ...  gs0  gs1  gs2  \
0        1    0    0    0    0    0    0    0    0    0  ...    0    0    0   
1        1    0    0    0    0    0    0    0    0    0  ...    0    0    0   
2        0    0    0    0    1    0    0    0    0    0  ...    0    0    0   
3        0    0    0    0    1    0    0    0    0    0  ...    0    0    0   
4        0    0    0    0    1    0    0    0    0    0  ...    0    0    0   
...    ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...  ...   
35578    0    0    0    0    0    0    0    0    0    0  ...    0    0    0   
35579    0    0    0    0    0    0    0    0    0    0  ...    0    0    0   
35580    0    0    0    0    0    0    0    0    0    0  ...    0    0    0   
35581    0    0    0    0    0    0    0    0    0    0  ...    0    0    0   
35582    0    0    0    0    0    0    0    0    0    0  ...    0    0    0   

       gs3  gs4  jur  pol  sap  sop  tb  
0        0    0    0    0    0    0   0  
1        0    0    0    0    0    0   0  
2        0    0    0    0    0    0   0  
3        0    0    0    0    0    0   0  
4        0    0    0    0    0    0   0  
...    ...  ...  ...  ...  ...  ...  ..  
35578    0    0    0    0    0    0   1  
35579    0    0    0    0    0    0   1  
35580    0    0    0    0    0    0   1  
35581    0    0    0    0    0    0   1  
35582    0    0    0    0    0    0   1  

[35583 rows x 31 columns]

In [29]:
#df.head()

In [30]:
df = df.reset_index().drop('index', axis=1)

In [31]:
df = pd.concat([df, df_types], axis=1)

In [32]:
df.head()

code_mission type_mission exercice code_client  total_assistant  \
0  AAGS2017010396900          AGS     2017        5782              2.0   
1  AAGS2017010397000          AGS     2017        6565              1.0   
2  ACOM2017007712400          COM     2017        6635              4.0   
3  ACOM2017007718600          COM     2017        6718              1.0   
4  ACOM2017007719900          COM     2017        6720              3.0   

  manager                                       code_article  \
0     JOP                                            ['029']   
1     FRL                                            ['001']   
2     GSO  ['010', '012', '001', '016', '011', '019', '013']   
3     DAC                                     ['011', '013']   
4     VAL                ['016', '001', '011', '013', '019']   

                                     libelle_article  temps  \
0                       ['AUTRES TRAVAUX EN SOCIAL']  43.00   
1                          ['RENSEIGNEMENTS CLIENT']   1.00   
2  ['SAISIE ET CENTRALISATION', 'DECLARATIONS T.V...  22.75   
3  ['CONTROLE ET REVISION', 'COMPTES ANNUELS  ET ...   7.50   
4  ['AUTRES DECLARATIONS FISCALES', 'RENSEIGNEMEN...  17.75   

   valorisation_temps  ... gs0  gs1  gs2  gs3 gs4 jur  pol  sap  sop  tb  
0             5231.25  ...   0    0    0    0   0   0    0    0    0   0  
1               68.00  ...   0    0    0    0   0   0    0    0    0   0  
2             1372.50  ...   0    0    0    0   0   0    0    0    0   0  
3              555.00  ...   0    0    0    0   0   0    0    0    0   0  
4             1168.75  ...   0    0    0    0   0   0    0    0    0   0  

[5 rows x 47 columns]

## Bag of Words sur les activités

In [33]:
## Création d'un "bag of words" pour analyser les données des libellés activités

In [34]:
def clean_act(s):
    s_lower = s.lower()
    s_sans_chiffre = re.sub("[0-9]","", s_lower)
    s_sans_par = re.sub(r'\([^)]*\)', '', s_sans_chiffre)
    s_sanscarac = re.sub('[^a-zA-Z\s,]', '', s_sans_par)
    return s_sanscarac

In [35]:
df["libelle_article"] = df.libelle_article.apply(clean_act)

In [36]:
df.head()

code_mission type_mission exercice code_client  total_assistant  \
0  AAGS2017010396900          AGS     2017        5782              2.0   
1  AAGS2017010397000          AGS     2017        6565              1.0   
2  ACOM2017007712400          COM     2017        6635              4.0   
3  ACOM2017007718600          COM     2017        6718              1.0   
4  ACOM2017007719900          COM     2017        6720              3.0   

  manager                                       code_article  \
0     JOP                                            ['029']   
1     FRL                                            ['001']   
2     GSO  ['010', '012', '001', '016', '011', '019', '013']   
3     DAC                                     ['011', '013']   
4     VAL                ['016', '001', '011', '013', '019']   

                                     libelle_article  temps  \
0                           autres travaux en social  43.00   
1                              renseignements client   1.00   
2  saisie et centralisation, declarations tva, re...  22.75   
3  controle et revision, comptes annuels  et decl...   7.50   
4  autres declarations fiscales, renseignements c...  17.75   

   valorisation_temps  ... gs0  gs1  gs2  gs3 gs4 jur  pol  sap  sop  tb  
0             5231.25  ...   0    0    0    0   0   0    0    0    0   0  
1               68.00  ...   0    0    0    0   0   0    0    0    0   0  
2             1372.50  ...   0    0    0    0   0   0    0    0    0   0  
3              555.00  ...   0    0    0    0   0   0    0    0    0   0  
4             1168.75  ...   0    0    0    0   0   0    0    0    0   0  

[5 rows x 47 columns]

In [37]:
text_act = np.array(df.libelle_article.values)
text_act

array(['autres travaux en social', 'renseignements client',
       'saisie et centralisation, declarations tva, renseignements client, autres declarations fiscales, controle et revision, presentation des comptes annuels, comptes annuels  et declarations fiscales',
       ..., 'tableau de bord', 'tableau de bord', 'tableau de bord'],
      dtype=object)

In [38]:
# Create the bag of words feature matrix
count = CountVectorizer(analyzer='word', tokenizer=lambda x: x.split(', '))

bow_act = count.fit_transform(text_act)

# Show feature matrix
bow_act.toarray()

array([[0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       ...,
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0]], dtype=int64)

In [39]:
feature_names = count.get_feature_names()

In [40]:
df_act = pd.DataFrame(bow_act.toarray(), columns=feature_names)
df_act

accompagnement dans le suivi de vos travaux  \
0                                                0   
1                                                0   
2                                                0   
3                                                0   
4                                                0   
...                                            ...   
35578                                            0   
35579                                            0   
35580                                            0   
35581                                            0   
35582                                            0   

       accompagnement fonction daf  \
0                                0   
1                                0   
2                                0   
3                                0   
4                                0   
...                            ...   
35578                            0   
35579                            0   
35580                            0   
35581                            0   
35582                            0   

       actualisation mensuelle aide a lembauche  apprenti  \
0                                                      0    
1                                                      0    
2                                                      0    
3                                                      0    
4                                                      0    
...                                                  ...    
35578                                                  0    
35579                                                  0    
35580                                                  0    
35581                                                  0    
35582                                                  0    

       aide a lembauche  apprenti  animation et formatioin  \
0                               0                        0   
1                               0                        0   
2                               0                        0   
3                               0                        0   
4                               0                        0   
...                           ...                      ...   
35578                           0                        0   
35579                           0                        0   
35580                           0                        0   
35581                           0                        0   
35582                           0                        0   

       assistance au controle fiscal  assistance au controle urssaf  \
0                                  0                              0   
1                                  0                              0   
2                                  0                              0   
3                                  0                              0   
4                                  0                              0   
...                              ...                            ...   
35578                              0                              0   
35579                              0                              0   
35580                              0                              0   
35581                              0                              0   
35582                              0                              0   

       assistance commissaire aux comptes  assistance informatique client  \
0                                       0                               0   
1                                       0                               0   
2                                       0                               0   
3                                       0                               0   
4                                       0                               0   
...                                   ...                             ...   
35578                          

In [41]:
df_act.columns

Index(['accompagnement dans le suivi de vos travaux',
       'accompagnement fonction daf',
       'actualisation mensuelle aide a lembauche  apprenti',
       'aide a lembauche  apprenti', 'animation et formatioin',
       'assistance au controle fiscal', 'assistance au controle urssaf',
       'assistance commissaire aux comptes', 'assistance informatique client',
       'assistance telephonique a lheure',
       ...
       'secretariat juridique annuel sceaearl',
       'secretariat juridique annuel scmsc',
       'secrtariat juridique annuel sci  sc portefeuille',
       'situation intermediaire', 'tableau de bord',
       'traitement activite partielle sur bulletin de paie',
       'transmission de lentreprise', 'travaux de fin de mission',
       'travaux exceptionnels covd', 'verification finale travaux'],
      dtype='object', length=173)

In [42]:
df = pd.concat([df, df_act], axis=1)

In [43]:
df.head()

code_mission type_mission exercice code_client  total_assistant  \
0  AAGS2017010396900          AGS     2017        5782              2.0   
1  AAGS2017010397000          AGS     2017        6565              1.0   
2  ACOM2017007712400          COM     2017        6635              4.0   
3  ACOM2017007718600          COM     2017        6718              1.0   
4  ACOM2017007719900          COM     2017        6720              3.0   

  manager                                       code_article  \
0     JOP                                            ['029']   
1     FRL                                            ['001']   
2     GSO  ['010', '012', '001', '016', '011', '019', '013']   
3     DAC                                     ['011', '013']   
4     VAL                ['016', '001', '011', '013', '019']   

                                     libelle_article  temps  \
0                           autres travaux en social  43.00   
1                              renseignements client   1.00   
2  saisie et centralisation, declarations tva, re...  22.75   
3  controle et revision, comptes annuels  et decl...   7.50   
4  autres declarations fiscales, renseignements c...  17.75   

   valorisation_temps  ... secretariat juridique annuel sceaearl  \
0             5231.25  ...                                     0   
1               68.00  ...                                     0   
2             1372.50  ...                                     0   
3              555.00  ...                                     0   
4             1168.75  ...                                     0   

   secretariat juridique annuel scmsc  \
0                                   0   
1                                   0   
2                                   0   
3                                   0   
4                                   0   

   secrtariat juridique annuel sci  sc portefeuille  situation intermediaire  \
0                                                 0                        0   
1                                                 0                        0   
2                                                 0                        0   
3                                                 0                        0   
4                                                 0                        0   

  tableau de bord traitement activite partielle sur bulletin de paie  \
0               0                                                  0   
1               0                                                  0   
2               0                                                  0   
3               0                                                  0   
4               0                                                  0   

   transmission de lentreprise  travaux de fin de mission  \
0                            0                          0   
1                            0                          0   
2                            0                          0   
3                            0                          0   
4                            0                          0   

   travaux exceptionnels covd  verification finale travaux  
0                           0                            0  
1                           0                            0  
2                           0                            0  
3                           0                            0  
4                           0                            0  

[5 rows x 220 columns]

In [44]:
df.describe()

total_assistant         temps  valorisation_temps    facturation  \
count     35583.000000  35583.000000        35583.000000   35583.000000   
mean          3.544642     45.219825         1656.745699    1595.180793   
std           2.079425    570.992930         2666.712615    3496.031262   
min           1.000000      0.000000           -1.500000    -600.000000   
25%           2.000000      3.500000          261.500000     360.000000   
50%           3.000000     10.000000          736.500000     670.000000   
75%           5.000000     33.250000         2036.200000    1905.000000   
max          25.000000  36328.500000        91573.750000  383660.000000   

         entrée_clt    sortie_clt           ags           cac           cap  \
count  35583.000000  35583.000000  35583.000000  35583.000000  35583.000000   
mean       0.089874      0.077397      0.000225      0.006098      0.000028   
std        0.286006      0.267223      0.014993      0.077855      0.005301   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        0.000000      0.000000      0.000000      0.000000      0.000000   
50%        0.000000      0.000000      0.000000      0.000000      0.000000   
75%        0.000000      0.000000      0.000000      0.000000      0.000000   
max        1.000000      1.000000      1.000000      1.000000      1.000000   

                cat  ...  secretariat juridique annuel sceaearl  \
count  35583.000000  ...                           35583.000000   
mean       0.000084  ...                               0.000393   
std        0.009182  ...                               0.019832   
min        0.000000  ...                               0.000000   
25%        0.000000  ...                               0.000000   
50%        0.000000  ...                               0.000000   
75%        0.000000  ...                               0.000000   
max        1.000000  ...                               1.000000   

       secretariat juridique annuel scmsc  \
count                        35583.000000   
mean                             0.000365   
std                              0.019111   
min                              0.000000   
25%                              0.000000   
50%                              0.000000   
75%                              0.000000   
max                              1.000000   

       secrtariat juridique annuel sci  sc portefeuille  \
count                                      35583.000000   
mean                                           0.015822   
std                                            0.124789   
min                                            0.000000   
25%                                            0.000000   
50%                                            0.000000   
75%                                            0.000000   
max                                            1.000000   

       situation intermediaire  tableau de bord  \
count             35583.000000     35583.000000   
mean                  0.014895         0.008909   
std                   0.121133         0.095450   
min                   0.000000         0.000000   
25%                   0.000000         0.000000   
50%                   0.000000         0.000000   
75%                   0.000000         0.000000   
max                   1.000000         2.000000   

       traitement activite partielle sur bulletin de paie  \
count                                       35583.000000    
mean                                            0.020544    
std                                             0.141852    
min                                             0.000000    
25%                                             0.000000    
50%                                             0.000000    
75%                                             0.000000    
max                                             1.000000    

       transmission de lentreprise  travaux de fin de mission  \
co

In [45]:
#df.columns[46:96]

In [46]:
#df.columns[96:146]

In [47]:
#df.columns[146:196]

In [48]:
#df.columns[196:]

In [49]:
#52 lignes comprennent de la sous-traitance
df[df['saisie primo bpo'] == 1]

code_mission type_mission exercice code_client  total_assistant  \
40     ACOM2017008059000          COM     2017       40082              4.0   
279    ACOM2017008223100          COM     2017        6129              3.0   
312    ACOM2017008227400          COM     2017       40032              5.0   
316    ACOM2017008227900          COM     2017       40066              7.0   
465    ACOM2017008247900          COM     2017        5954              3.0   
467    ACOM2017008248100          COM     2017        5956              7.0   
473    ACOM2017008249100          COM     2017        6139              5.0   
1196   ACOM2017008367900          COM     2017        6944              3.0   
1227   ACOM2017008488000          COM     2017        6988              4.0   
1332   ACOM2017008594000          COM     2017        7135              7.0   
1333   ACOM2017008594300          COM     2017        7136              7.0   
1388   ACOM2017008687700          COM     2017        7498              4.0   
1481   ACOM2017008958700          COM     2017        7508              4.0   
1914   ACOM2017009196600          COM     2017        5036              7.0   
1939   ACOM2017009199800          COM     2017        5159              5.0   
1971   ACOM2017009203700          COM     2017        5425              5.0   
2007   ACOM2017009207800          COM     2017        5563              5.0   
2142   ACOM2017009222900          COM     2017        6123              4.0   
2143   ACOM2017009223000          COM     2017        6124              4.0   
2159   ACOM2017009225200          COM     2017        6209              5.0   
2163   ACOM2017009225600          COM     2017        6248              2.0   
2230   ACOM2017009234200          COM     2017        6573              5.0   
2231   ACOM2017009234300          COM     2017        6574              5.0   
2278   ACOM2017009240500          COM     2017        6757              7.0   
2481   ACOM2017009423400          COM     2017        7910              8.0   
2533   ACOM2017009555500          COM     2017        7986              6.0   
2539   ACOM2017009560700          COM     2017        7968              6.0   
2960   ACOM2017010140600          COM     2017       A0520              5.0   
3407   ACOM2017010581200          COM     2017       A0651              9.0   
7900   ACOM2018008594100          COM     2018        7135              7.0   
7901   ACOM2018008594400          COM     2018        7136              7.0   
7926   ACOM2018008687800          COM     2018        7498              4.0   
7942   ACOM2018008713200          COM     2018       40082              4.0   
8168   ACOM2018008769200          COM     2018        6129              5.0   
8259   ACOM2018008958800          COM     2018        7508              5.0   
9147   ACOM2018009246700          COM     2018        6944              6.0   
9163   ACOM2018009248800          COM     2018        6988              4.0   
9627   ACOM2018009329100          COM     2018        5425              6.0   
9662   ACOM2018009333200          COM     2018        5563              5.0   
9779   ACOM2018009348200          COM     2018        6123              7.0   
9780   ACOM2018009348300          COM     2018        6124              6.0   
9856   ACOM2018009359400          COM     2018        6573              5.0   
9857   ACOM2018009359500          COM     2018        6574              6.0   
9900   ACOM2018009365700          COM     2018        6757              5.0   
10083  ACOM2018009423500          COM     2018        7910              6.0   
10129  ACOM2018009560800          COM     2018        7968              6.0   
10450  ACOM2018010140700          COM     2018       A0520              8.0   
10920  ACOM2018010581300          COM     2018       A0651              6.0   
11119  ACOM2018011495900          COM     2018       A0922              4.0   
11233  ACOM2018011816300          COM     2018       A1063    

### Création de familles d'activités pour réduire le nombre de colonnes

In [50]:
#Création famille assistance téléphonique
df['Assistance téléphonique-hotline'] =  df['assistance telephonique a lheure'] + df['assistance tlphonique  lheure'] + df['assistance tlphonique annuelle '] + df['forfait assistance telephonique'] + df['hotline utilisation logiciel grh']


In [51]:
#Remplacer les autres valeurs par 1 ou 0
df['Assistance téléphonique-hotline'] = df['Assistance téléphonique-hotline'].map({0:0, 1:1, 2:1, 3:1})

In [52]:
#Supprimer les colonnes agrégées dans la famille d'activité
df.drop(['assistance telephonique a lheure', 'assistance tlphonique  lheure', 'assistance tlphonique annuelle ', 'forfait assistance telephonique', 'hotline utilisation logiciel grh'], axis=1, inplace=True)

In [53]:
#Création famille Autres travaux exceptionnels
df['Autres travaux exceptionnels'] =  df['autres travaux'] + df['autres travaux exceptionnels']

In [54]:
df[(df['Autres travaux exceptionnels'] != 1) & (df['Autres travaux exceptionnels'] != 0)]

Empty DataFrame
Columns: [code_mission, type_mission, exercice, code_client, total_assistant, manager, code_article, libelle_article, temps, valorisation_temps, site, facturation, entrée_clt, sortie_clt, secteur, forme, ags, cac, cap, cat, com, ge0, ge1, ge2, ge3, gec, gej, gep, ges, gj0, gj1, gj2, gj3, gj4, gj5, gj6, gj7, gs0, gs1, gs2, gs3, gs4, jur, pol, sap, sop, tb, accompagnement dans le suivi de vos travaux, accompagnement fonction daf, actualisation mensuelle aide a lembauche  apprenti, aide a lembauche  apprenti, animation et formatioin, assistance au controle fiscal, assistance au controle urssaf, assistance commissaire aux comptes, assistance informatique client, attestations particulires, audit  diagnostic, audit bulletins de paie etou charges sociales, audit des comptes  rdaction des rapports, audit interne dossiers, audit social, autres declarations fiscales, autres missions commissariat, autres prestations sociales, autres travaux, autres travaux en social, autres travaux exceptionnels, avenant au contrat de travail, avertissement, bilan retraite, budget previsionnel, bulletin de paie, bulletin de paie apprenti, bulletin de paie cddmoisextrasaisonnier, bulletin de paie clarifie, bulletin de paie refait, bulletin de paie simulation, bulletin de paie suite prudhomme, calcul cout rupture de contrat, cerfa apprentissage, cerfa contrat de professionalisation, charte teletravail, circularisation, coffre fort numerique digiiposte par salarie, coffre fort numerique digiposte  forfait mise en route, collecte des lments  inventaire, comptes annuels  et declarations fiscales, contrat de travail, contrat de travail specifique, contrats de travail, contrle des charges, contrle des comptes, contrle interne, controle charges sociales, controle de charges, controle et revision, convention abandon de compte courant, convention de mutation, courrier reponse salarie, ...]
Index: []

[0 rows x 217 columns]

In [55]:
#Supprimer les colonnes agrégées dans la famille d'activité
df.drop(['autres travaux', 'autres travaux exceptionnels'], axis=1, inplace=True)

In [56]:
#Création famille gestion paie
df['Gestion bulletins de paie'] =  df['bulletin de paie'] + df['bulletin de paie apprenti'] + df['bulletin de paie cddmoisextrasaisonnier'] + df['bulletin de paie clarifie'] + df['bulletin de paie refait'] + df['bulletin de paie simulation'] + df['bulletin de paie suite prudhomme'] + df['fiche de paie'] + df['remontee des bulletins de paie'] + df['remontee des bulletins de salaire existants avec modifications'] + df['remontee des bulletins de salaire existants sans modifications']

In [57]:
#df[(df['Gestion bulletins de paie'] != 1) & (df['Gestion bulletins de paie'] != 0)]

In [58]:
#Remplacer les autres valeurs par 1 ou 0
df['Gestion bulletins de paie'] = df['Gestion bulletins de paie'].map({0:0, 1:1, 2:1, 3:1, 4:1, 5:1, 6:1})

In [59]:
df[(df['Gestion bulletins de paie'] != 1) & (df['Gestion bulletins de paie'] != 0)]

Empty DataFrame
Columns: [code_mission, type_mission, exercice, code_client, total_assistant, manager, code_article, libelle_article, temps, valorisation_temps, site, facturation, entrée_clt, sortie_clt, secteur, forme, ags, cac, cap, cat, com, ge0, ge1, ge2, ge3, gec, gej, gep, ges, gj0, gj1, gj2, gj3, gj4, gj5, gj6, gj7, gs0, gs1, gs2, gs3, gs4, jur, pol, sap, sop, tb, accompagnement dans le suivi de vos travaux, accompagnement fonction daf, actualisation mensuelle aide a lembauche  apprenti, aide a lembauche  apprenti, animation et formatioin, assistance au controle fiscal, assistance au controle urssaf, assistance commissaire aux comptes, assistance informatique client, attestations particulires, audit  diagnostic, audit bulletins de paie etou charges sociales, audit des comptes  rdaction des rapports, audit interne dossiers, audit social, autres declarations fiscales, autres missions commissariat, autres prestations sociales, autres travaux en social, avenant au contrat de travail, avertissement, bilan retraite, budget previsionnel, bulletin de paie, bulletin de paie apprenti, bulletin de paie cddmoisextrasaisonnier, bulletin de paie clarifie, bulletin de paie refait, bulletin de paie simulation, bulletin de paie suite prudhomme, calcul cout rupture de contrat, cerfa apprentissage, cerfa contrat de professionalisation, charte teletravail, circularisation, coffre fort numerique digiiposte par salarie, coffre fort numerique digiposte  forfait mise en route, collecte des lments  inventaire, comptes annuels  et declarations fiscales, contrat de travail, contrat de travail specifique, contrats de travail, contrle des charges, contrle des comptes, contrle interne, controle charges sociales, controle de charges, controle et revision, convention abandon de compte courant, convention de mutation, courrier reponse salarie, cration  migration silae, cration de dossier, ...]
Index: []

[0 rows x 216 columns]

In [60]:
#Supprimer les colonnes agrégées dans la famille d'activité
df.drop(['bulletin de paie','bulletin de paie apprenti', 'bulletin de paie cddmoisextrasaisonnier', 'bulletin de paie clarifie', 'bulletin de paie refait', 'bulletin de paie simulation', 'bulletin de paie suite prudhomme', 'fiche de paie', 'remontee des bulletins de paie', 'remontee des bulletins de salaire existants avec modifications', 'remontee des bulletins de salaire existants sans modifications'], axis=1, inplace=True)

In [61]:
#Création famille coffre fort numérique
df['Coffre fort numérique'] =  df['coffre fort numerique digiiposte par salarie'] + df['coffre fort numerique digiposte  forfait mise en route']

In [62]:
#Remplacer les autres valeurs par 1 ou 0
df['Coffre fort numérique'] = df['Coffre fort numérique'].map({0:0, 1:1, 2:1})

In [63]:
df[(df['Gestion bulletins de paie'] != 1) & (df['Gestion bulletins de paie'] != 0)]

Empty DataFrame
Columns: [code_mission, type_mission, exercice, code_client, total_assistant, manager, code_article, libelle_article, temps, valorisation_temps, site, facturation, entrée_clt, sortie_clt, secteur, forme, ags, cac, cap, cat, com, ge0, ge1, ge2, ge3, gec, gej, gep, ges, gj0, gj1, gj2, gj3, gj4, gj5, gj6, gj7, gs0, gs1, gs2, gs3, gs4, jur, pol, sap, sop, tb, accompagnement dans le suivi de vos travaux, accompagnement fonction daf, actualisation mensuelle aide a lembauche  apprenti, aide a lembauche  apprenti, animation et formatioin, assistance au controle fiscal, assistance au controle urssaf, assistance commissaire aux comptes, assistance informatique client, attestations particulires, audit  diagnostic, audit bulletins de paie etou charges sociales, audit des comptes  rdaction des rapports, audit interne dossiers, audit social, autres declarations fiscales, autres missions commissariat, autres prestations sociales, autres travaux en social, avenant au contrat de travail, avertissement, bilan retraite, budget previsionnel, calcul cout rupture de contrat, cerfa apprentissage, cerfa contrat de professionalisation, charte teletravail, circularisation, coffre fort numerique digiiposte par salarie, coffre fort numerique digiposte  forfait mise en route, collecte des lments  inventaire, comptes annuels  et declarations fiscales, contrat de travail, contrat de travail specifique, contrats de travail, contrle des charges, contrle des comptes, contrle interne, controle charges sociales, controle de charges, controle et revision, convention abandon de compte courant, convention de mutation, courrier reponse salarie, cration  migration silae, cration de dossier, decision unilaterale employeur, declaration annuelle dads, declaration arret de travail, declaration des beneficiaires effectifs, declaration des beneficiaires effectifs suivants, declarations mensuelles  trimestrielles, declarations tva, ...]
Index: []

[0 rows x 206 columns]

In [64]:
#Supprimer les colonnes agrégées dans la famille d'activité
df.drop(['coffre fort numerique digiiposte par salarie', 'coffre fort numerique digiposte  forfait mise en route'], axis=1, inplace=True)

In [65]:
#Création famille assistance téléphonique
df['contrat de travail et/ou avenant'] =  df['contrat de travail'] + df['contrat de travail specifique'] + df['contrats de travail'] + df['avenant au contrat de travail']

In [66]:
df[(df['contrat de travail et/ou avenant'] != 1) & (df['contrat de travail et/ou avenant'] != 0)]

code_mission type_mission exercice code_client  total_assistant  \
4489   AGES2017010134700          GES     2017        7406              5.0   
4490   AGES2017010135300          GES     2017        7402              6.0   
4497   AGES2017010139300          GES     2017        7407              6.0   
4508   AGES2017010226000          GES     2017        7408              3.0   
4509   AGES2017010226100          GES     2017        7985              4.0   
...                  ...          ...      ...         ...              ...   
21958  AGS02019012616400          GS0     2019        7407              4.0   
21974  AGS02019012630400          GS0     2019        7192              5.0   
22018  AGS02019012730300          GS0     2019        7225              3.0   
22024  AGS02019012738000          GS0     2019       A0883              5.0   
22070  AGS02019012796800          GS0     2019        7388              2.0   

      manager                                       code_article  \
4489      THB    ['025', '024', 'F025', 'F0251', '029', 'F0242']   
4490      THB  ['025', '029', 'F025', 'F0251', '027', '024', ...   
4497      THB  ['025', '029', 'F025', 'F0243', 'F0242', '024'...   
4508      THB                                    ['025', 'F025']   
4509      EMG             ['025', '029', 'F0251', '024', 'F025']   
...       ...                                                ...   
21958     THB                                   ['025', 'F0251']   
21974     EMG                          ['025', 'F025', 'F02504']   
22018     EMG                             ['025', 'F025', '024']   
22024     EMG                             ['025', '001', 'F025']   
22070     THB                                    ['F025', '025']   

                                         libelle_article   temps  \
4489   contrats de travail, formalites de licenciemen...   54.25   
4490   contrats de travail, autres travaux en social,...  121.50   
4497   contrats de travail, autres travaux en social,...   45.00   
4508             contrats de travail, contrat de travail    2.75   
4509   contrats de travail, autres travaux en social,...   11.25   
...                                                  ...     ...   
21958  contrats de travail, avenant au contrat de tra...    4.25   
21974  contrats de travail, contrat de travail, cerfa...   17.25   
22018  contrats de travail, contrat de travail, forma...    5.75   
22024  contrats de travail, renseignements client, co...   12.00   
22070            contrat de travail, contrats de travail    2.25   

       valorisation_temps  ...  \
4489              3881.00  ...   
4490              9030.75  ...   
4497              3331.75  ...   
4508               239.50  ...   
4509               809.75  ...   
...                   ...  ...   
21958              255.25  ...   
21974             1319.75  ...   
22018              452.50  ...   
22024             1027.63  ...   
22070              182.50  ...   

      traitement activite partielle sur bulletin de paie  \
4489                                                   0   
4490                                                   0   
4497                                                   0   
4508                                                   0   
4509                                                   0   
...                                                  ...   
21958                                                  0   
21974                                                  0   
22018                                                  0   
22024                                                  0   
22070                                                  0   

       transmission de lentreprise  travaux de fin de mission  \
4489                             0                          0   
4490                             0                          0   
4497                             0                          0   
4508                             0     

In [67]:
#Remplacer les autres valeurs par 1 ou 0
df['contrat de travail et/ou avenant'] = df['contrat de travail et/ou avenant'].map({0:0, 1:1, 2:1, 3:1})

In [68]:
#Supprimer les colonnes agrégées dans la famille d'activité
df.drop(['contrat de travail', 'contrat de travail specifique', 'contrats de travail', 'avenant au contrat de travail'], axis=1, inplace=True)

In [69]:
#Création famille contrôle des charges
df['Contrôle des charges'] =  df['contrle des charges'] + df['controle charges sociales'] + df['controle de charges']

In [70]:
df[(df['Contrôle des charges'] != 1) & (df['Contrôle des charges'] != 0)]

code_mission type_mission exercice code_client  total_assistant  \
24396  APOL2019012180700          POL     2019        7315              9.0   
24436  APOL2019012870300          POL     2019       A1345              3.0   
24437  APOL2019012870400          POL     2019       A1346              3.0   

      manager                                       code_article  \
24396     NAB  ['F0304', '0304', '0310', '0302', '029', 'F031...   
24436     EMG          ['F0300', '0310', '0317', 'F0317', '029']   
24437     EMG  ['F0300', '0303', '0317', '0310', '0316', '029...   

                                         libelle_article   temps  \
24396  assistance telephonique a lheure, assistance t...  578.25   
24436  mise a disposition du logiciel de paie, mise  ...   10.50   
24437  mise a disposition du logiciel de paie, format...  139.25   

       valorisation_temps  ... transmission de lentreprise  \
24396             8811.31  ...                           0   
24436             1240.00  ...                           0   
24437             1665.00  ...                           0   

       travaux de fin de mission  travaux exceptionnels covd  \
24396                          0                           0   
24436                          0                           0   
24437                          0                           0   

       verification finale travaux Assistance téléphonique-hotline  \
24396                            0                               1   
24436                            0                               0   
24437                            0                               1   

      Autres travaux exceptionnels  Gestion bulletins de paie  \
24396                            0                          0   
24436                            0                          0   
24437                            0                          0   

       Coffre fort numérique  contrat de travail et/ou avenant  \
24396                      0                                 0   
24436                      0                                 0   
24437                      0                                 0   

       Contrôle des charges  
24396                     2  
24436                     2  
24437                     2  

[3 rows x 202 columns]

In [71]:
#Remplacer les autres valeurs par 1 ou 0
df['Contrôle des charges'] = df['Contrôle des charges'].map({0:0, 1:1, 2:1})

In [72]:
#Supprimer les colonnes agrégées dans la famille d'activité
df.drop(['contrle des charges', 'controle charges sociales', 'controle de charges'], axis=1, inplace=True)

In [73]:
#Création famille déclaration des bénéficiaires effectifs
df['Déclaration des bénéficiaires effectifs'] =  df['declaration des beneficiaires effectifs'] + df['declaration des beneficiaires effectifs suivants']

In [74]:
df[(df['Déclaration des bénéficiaires effectifs'] != 1) & (df['Déclaration des bénéficiaires effectifs'] != 0)]

code_mission type_mission exercice code_client  total_assistant  \
4050   AGEJ2017010544700          GEJ     2017        7551              4.0   
4075   AGEJ2017010577900          GEJ     2017       60248              6.0   
4179   AGEJ2017010678000          GEJ     2017        7251              3.0   
4258   AGEJ2017010804200          GEJ     2017       A0315              2.0   
4260   AGEJ2017010804700          GEJ     2017        4895              2.0   
...                  ...          ...      ...         ...              ...   
12347  AGEJ2018011725200          GEJ     2018        7052              3.0   
12348  AGEJ2018011725300          GEJ     2018        7159              8.0   
12353  AGEJ2018011725800          GEJ     2018       A0280              2.0   
12354  AGEJ2018011725900          GEJ     2018       A0309              2.0   
12355  AGEJ2018011726000          GEJ     2018       A0434              2.0   

                     manager                           code_article  \
4050                     NAB  ['069', 'F06453', 'F06454', 'F06455']   
4075   _OLP NE PLUS UTILISER     ['065', '069', 'F06454', 'F06455']   
4179                     NAB            ['069', 'F06454', 'F06455']   
4258                     JOP            ['069', 'F06454', 'F06455']   
4260                     JOP  ['069', 'F06453', 'F06454', 'F06455']   
...                      ...                                    ...   
12347                    VAL            ['F06454', 'F06455', '069']   
12348                    DAC     ['F06454', 'F06455', '069', '065']   
12353                    THB            ['F06454', 'F06455', '069']   
12354                    EMG            ['F06454', 'F06455', '069']   
12355                    GSO            ['069', 'F06454', 'F06455']   

                                         libelle_article  temps  \
4050   divers juridique, ouverture et paraphe registr...   7.50   
4075   formalites de dissolution  et de liquidation, ...  10.00   
4179   divers juridique, declaration des beneficiaire...   4.75   
4258   divers juridique, declaration des beneficiaire...   3.50   
4260   divers juridique, ouverture et paraphe registr...   4.25   
...                                                  ...    ...   
12347  declaration des beneficiaires effectifs, decla...   2.75   
12348  declaration des beneficiaires effectifs, decla...  18.25   
12353  declaration des beneficiaires effectifs, decla...   2.25   
12354  declaration des beneficiaires effectifs, decla...   2.25   
12355  divers juridique, declaration des beneficiaire...   3.50   

       valorisation_temps  ... travaux de fin de mission  \
4050               430.25  ...                         0   
4075               667.25  ...                         0   
4179               257.00  ...                         0   
4258               189.50  ...                         0   
4260               215.00  ...                         0   
...                   ...  ...                       ...   
12347              136.75  ...                         0   
12348             1254.75  ...                         0   
12353              110.75  ...                         0   
12354              110.75  ...                         0   
12355              197.00  ...                         0   

       travaux exceptionnels covd  verification finale travaux  \
4050                            0                            0   
4075                            0                            0   
4179                            0                            0   
4258                            0                            0   
4260                            0                            0   
...                           ...                          ...   
12347                           0                            0   
12348                           0                            0   
12353                           0                            0   
12354                   

In [75]:
#Remplacer les autres valeurs par 1 ou 0
df['Déclaration des bénéficiaires effectifs'] = df['Déclaration des bénéficiaires effectifs'].map({0:0, 1:1, 2:1})

In [76]:
#Supprimer les colonnes agrégées dans la famille d'activité
df.drop(['declaration des beneficiaires effectifs', 'declaration des beneficiaires effectifs suivants'], axis=1, inplace=True)

In [77]:
#Création formalites de rupture de contrat de travail
df['Formalités de rupture de contrat de travail'] =  df['calcul cout rupture de contrat'] + df['formalites de rupture conventionnelle'] + df['formalites de licenciement'] + df['etablissement attestation pole emploi'] + df['mouvement de personnel sortie'] + df['mouvement de personnel sortie apprenti'] + df['mouvement de personnel sortie cddmoisextrasaisonnier']

In [78]:
df[(df['Formalités de rupture de contrat de travail'] != 1) & (df['Formalités de rupture de contrat de travail'] != 0)]

code_mission type_mission exercice code_client  total_assistant  \
4497   AGES2017010139300          GES     2017        7407              6.0   
4553   AGES2017010392400          GES     2017        7232              3.0   
6624   ASOP2017010108000          SOP     2017       A0520             11.0   
6626   ASOP2017010111600          SOP     2017       A0015              6.0   
6628   ASOP2017010117300          SOP     2017       A0532              4.0   
...                  ...          ...      ...         ...              ...   
35465  ASOP2020013189600          SOP     2020        2547              5.0   
35473  ASOP2020013221800          SOP     2020       A0647             12.0   
35491  ASOP2020013282500          SOP     2020        3979              8.0   
35494  ASOP2020013332200          SOP     2020       A2142             11.0   
35507  ASOP2020013403400          SOP     2020       A1891              7.0   

      manager                                       code_article  \
4497      THB  ['025', '029', 'F025', 'F0243', 'F0242', '024'...   
4553      EMG    ['024', '029', '025', 'F0241', 'F028', 'F0253']   
6624      DAC  ['030', '029', '020', 'F0200', 'F0211', 'F0206...   
6626      NAB  ['F0200', 'F0214', '020', '021', 'F0206', '029...   
6628      JOP  ['030', 'F0200', '029', 'F0211', '020', '021',...   
...       ...                                                ...   
35465     GSO  ['001', '029', '030', 'F0207.1', 'F0205', 'F02...   
35473     THB  ['F0201', 'F0205', 'F02082', 'F0200', '029', '...   
35491     CBA  ['029', '04950', 'F02083', 'F0228', 'F0200', '...   
35494     JOP  ['029', '030', 'F0207', 'F0205', 'F02081', '02...   
35507     JOD  ['030', '029', '020', '0291', '001', 'F0207', ...   

                                         libelle_article   temps  \
4497   contrats de travail, autres travaux en social,...   45.00   
4553   formalites de licenciement, autres travaux en ...   10.50   
6624   cration  migration silae, autres travaux en so...  314.50   
6626   bulletin de paie, actualisation mensuelle aide...   69.50   
6628   cration  migration silae, bulletin de paie, au...   84.00   
...                                                  ...     ...   
35465  renseignements client, autres travaux en socia...   33.50   
35473  bulletin de paie simulation, mouvement de pers...  282.75   
35491  autres travaux en social, travaux exceptionnel...  348.50   
35494  autres travaux en social, cration  migration s...   98.00   
35507  cration  migration silae, autres travaux en so...   77.50   

       valorisation_temps  ... travaux exceptionnels covd  \
4497              3331.75  ...                          0   
4553              1245.25  ...                          0   
6624             10370.75  ...                          0   
6626              3374.75  ...                          0   
6628              3673.50  ...                          0   
...                   ...  ...                        ...   
35465             2018.03  ...                          1   
35473            11400.42  ...                          1   
35491            12351.66  ...                          1   
35494             5256.61  ...                          0   
35507             4260.99  ...                          1   

       verification finale travaux  Assistance téléphonique-hotline  \
4497                             0                                0   
4553                             0                                0   
6624                             0                                0   
6626                             0                                0   
6628                             0                                0   
...                            ...                              ...   
35465                            0                                0   
35473                            0                                0   
35491                            0                       

In [79]:
#Remplacer les autres valeurs par 1 ou 0
df['Formalités de rupture de contrat de travail'] = df['Formalités de rupture de contrat de travail'].map({0:0, 1:1, 2:1, 3:1, 4:1})

In [80]:
#Supprimer les colonnes agrégées dans la famille d'activité
df.drop(['calcul cout rupture de contrat', 'formalites de rupture conventionnelle', 'formalites de licenciement', 'etablissement attestation pole emploi', 'mouvement de personnel sortie', 'mouvement de personnel sortie apprenti', 'mouvement de personnel sortie cddmoisextrasaisonnier'], axis=1, inplace=True)

In [81]:
#Création famille aide à l'embauche apprentissage
df["Aide à l'embauche apprentissage"] =  df['actualisation mensuelle aide a lembauche  apprenti'] + df['aide a lembauche  apprenti']

In [82]:
#df[(df["Aide à l'embauche apprentissage"] != 1) & (df["Aide à l'embauche apprentissage"] != 0)]

In [83]:
#Remplacer les autres valeurs par 1 ou 0
df["Aide à l'embauche apprentissage"] = df["Aide à l'embauche apprentissage"].map({0:0, 1:1, 2:1})

In [84]:
#Supprimer les colonnes agrégées dans la famille d'activité
df.drop(['actualisation mensuelle aide a lembauche  apprenti', 'aide a lembauche  apprenti'], axis=1, inplace=True)

In [85]:
#Création famille formation client
df['Formation client'] =  df['animation et formatioin'] + df['formation technique a lutilisation du logiciel'] + df['formation technique utilisation logiciel'] + df['formation techniques de paie appliques sur silae'] + df['formation tl pas']

In [86]:
df[(df['Formation client'] != 1) & (df['Formation client'] != 0)]

code_mission type_mission exercice code_client  total_assistant  \
6580   APOL2017010389800          POL     2017        7645              8.0   
6599   APOL2017010665600          POL     2017       A0735              3.0   
6600   APOL2017010673100          POL     2017       60070              5.0   
6601   APOL2017010673600          POL     2017       A0745              5.0   
6602   APOL2017010766700          POL     2017        4799              3.0   
6603   APOL2017010774300          POL     2017        7315              3.0   
15591  APOL2018011433200          POL     2018       60349              5.0   
15594  APOL2018011446400          POL     2018       11871              3.0   
15605  APOL2018011474500          POL     2018        3262              6.0   
15611  APOL2018011475100          POL     2018        3979              6.0   
15652  APOL2018011617100          POL     2018        6639              4.0   
15655  APOL2018011707900          POL     2018        1906              2.0   
15656  APOL2018011743100          POL     2018        4505              4.0   
15659  APOL2018011936400          POL     2018         713              5.0   
24352  APOL2019012122800          POL     2019         676              9.0   
24355  APOL2019012150600          POL     2019       40117              5.0   
24396  APOL2019012180700          POL     2019        7315              9.0   
24432  APOL2019012184300          POL     2019        4897              4.0   
24433  APOL2019012298300          POL     2019       A1443              5.0   
24434  APOL2019012573200          POL     2019        3670              5.0   
24435  APOL2019012584400          POL     2019        5644              4.0   
33352  APOL2020012826000          POL     2020        4696              4.0   
33355  APOL2020013001800          POL     2020       A2022              4.0   
33357  APOL2020013003500          POL     2020        4754              2.0   
33435  APOL2020013067000          POL     2020       A1986              3.0   
33441  APOL2020013244000          POL     2020       A1142              2.0   
33442  APOL2020013539200          POL     2020       A2975              3.0   
33443  APOL2020013582600          POL     2020        7984              3.0   

      manager                                       code_article  \
6580      DAC  ['0301', '0304', 'F0304', 'F0300', '0303', '03...   
6599      JOP  ['0302', 'F0318', 'F03112', 'F0312', 'F0313', ...   
6600      JOP  ['F0318', '0303', '0304', 'F0304', 'F0300', '0...   
6601      VAL  ['0302', 'F0300', '0304', 'F0304', '0303', 'F0...   
6602      GSO  ['0303', 'F0318', 'F03113', 'F0312', 'F0313', ...   
6603      NAB        ['F0300', 'F0318', '0303', '0304', 'F0304']   
15591     NAB  ['0302', '0304', 'F0318', '0303', 'F0304', 'F0...   
15594     CIT  ['0302', 'F0318', '0303', 'F0300', 'F0216', '0...   
15605     CIT  ['F0304', 'F0216', 'F0300', '0304', '029', '03...   
15611     CBA  ['F0304', 'F0300', 'F0216', '0302', '0304', '0...   
15652     DAC  ['F0318', '0303', '0302', 'F0300', '0304', 'F0...   
15655     VAL  ['0302', '029', '0303', 'F0318', 'F0319', 'F03...   
15656     GSO  ['0303', 'F0318', 'F0319', 'F0300', '0304', 'F...   
15659     GSO  ['029', 'F0318', '0303', '0304', 'F0304', 'F03...   
24352     GSO  ['029', '0302', '0303', 'F0318', '0310', '0304...   
24355     DAC  ['0302', '0310', 'F0300', 'F03111', 'F0312', '...   
24396     NAB  ['F0304', '0304', '0310', '0302', '029', 'F031...   
24432     SLE  ['0302', 'F0310', 'F0300', 'F03113', 'F0312', ...   
24433     GSO  ['0303', 'F0318', 'F0300', 'F03111', 'F0312', ...   
24434     FRL  ['F0300', '0303', 'F0318', '0304', 'F0304', '0...   
24435     GSO  ['F03113', 'F0313', 'F0312', '0302', '0303', '...   
33352     DAC  ['0302', '029', '0304', 'F0304', 'F0318', '031...   
33355     STL  ['0302', 'F0318', '0318', '029', 'F0304', '030...   
33357     GSO  ['0302', 'F0318', '0318', 'F0300', 'F03112', '...   
33435     SLE  ['030

In [87]:
#Remplacer les autres valeurs par 1 ou 0
df["Formation client"] = df['Formation client'].map({0:0, 1:1, 2:1, 3:1})

In [88]:
#Supprimer les colonnes agrégées dans la famille d'activité
df.drop(['animation et formatioin', 'formation technique a lutilisation du logiciel', 'formation technique utilisation logiciel', 'formation techniques de paie appliques sur silae', 'formation tl pas'], axis=1, inplace=True)

In [89]:
#Création famille secrétariat juridique
df['Secrétariat juridique'] =  df['secretariat juridique annuel'] + df['secretariat juridique annuel eurl'] + df['secretariat juridique annuel sa'] + df['secretariat juridique annuel sas'] + df['secretariat juridique annuel sasu'] + df['secretariat juridique annuel sceaearl'] + df['secretariat juridique annuel scmsc'] + df['secrtariat juridique annuel sci  sc portefeuille'] + df['divers juridique'] + df['juridique'] + df['secretariat juridique annuel sarl  snc']

In [90]:
df[(df['Secrétariat juridique'] != 1) & (df['Secrétariat juridique'] != 0)]

code_mission type_mission exercice code_client  total_assistant  \
740    ACOM2017008290300          COM     2017        5605              5.0   
1100   ACOM2017008337600          COM     2017       40138              3.0   
1148   ACOM2017008343400          COM     2017       60248              9.0   
1903   ACOM2017009195400          COM     2017        4951             10.0   
3041   ACOM2017010154200          COM     2017        7829             12.0   
...                  ...          ...      ...         ...              ...   
33301  AJUR2020013246600          JUR     2020       A0852              3.0   
33304  AJUR2020013276400          JUR     2020       A2103              3.0   
33312  AJUR2020013318700          JUR     2020       A1248              4.0   
33331  AJUR2020013561800          JUR     2020       A1041              1.0   
33349  AJUR2020014858700          JUR     2020       A1190              3.0   

                     manager  \
740                      VAL   
1100                     DAC   
1148   _OLP NE PLUS UTILISER   
1903   _OLP NE PLUS UTILISER   
3041                     THB   
...                      ...   
33301                    FRL   
33304                    EMG   
33312                    SLE   
33331                    EMG   
33349                    DAC   

                                            code_article  \
740           ['069', '001', '060', '016', '011', '013']   
1100   ['016', '012', '001', '060', '010', '049', '06...   
1148   ['069', '060', '001', '016', '012', '013', '011']   
1903   ['011', '010', '001', '012', '016', '018', '06...   
3041   ['001', '010', '041', '060', '049', '012', '01...   
...                                                  ...   
33301                                     ['069', '060']   
33304                                     ['060', '069']   
33312                                     ['060', '069']   
33331                                     ['060', '069']   
33349                                 ['060', '069', 60]   

                                         libelle_article   temps  \
740    divers juridique, renseignements client, secre...    8.75   
1100   autres declarations fiscales, declarations tva...   20.25   
1148   divers juridique, secretariat juridique annuel...   15.00   
1903   controle et revision, saisie et centralisation...  169.25   
3041   renseignements client, saisie et centralisatio...  167.00   
...                                                  ...     ...   
33301     divers juridique, secretariat juridique annuel    3.50   
33304     secretariat juridique annuel, divers juridique    4.75   
33312     secretariat juridique annuel, divers juridique    4.75   
33331     secretariat juridique annuel, divers juridique    6.00   
33349     secretariat juridique annuel, divers juridique    3.75   

       valorisation_temps  ... Autres travaux exceptionnels  \
740                514.25  ...                            0   
1100              1626.00  ...                            1   
1148              1068.75  ...                            0   
1903              8891.00  ...                            0   
3041             10277.25  ...                            1   
...                   ...  ...                          ...   
33301              237.50  ...                            0   
33304              324.64  ...                            0   
33312              301.89  ...                            0   
33331              489.00  ...                            0   
33349              256.50  ...                            0   

       Gestion bulletins de paie  Coffre fort numérique  \
740                            0                      0   
1100                           0                      0   
1148                           0                      0   
1903                           0                      0   
3041                           0                      0   
...                          

In [91]:
#Remplacer les autres valeurs par 1 ou 0
df["Secrétariat juridique"] = df['Secrétariat juridique'].map({0:0, 1:1, 2:1, 3:1})

In [92]:
#Supprimer les colonnes agrégées dans la famille d'activité
df.drop(['secretariat juridique annuel', 'secretariat juridique annuel eurl', 'secretariat juridique annuel sa', 'secretariat juridique annuel sas', 'secretariat juridique annuel sasu', 'secretariat juridique annuel sceaearl','secretariat juridique annuel scmsc','secrtariat juridique annuel sci  sc portefeuille', 'divers juridique', 'juridique'], axis=1, inplace=True)

In [93]:
#Création famille Audit social
df['Audit social'] =  df['audit bulletins de paie etou charges sociales'] + df['audit social']

In [94]:
df[(df['Audit social'] != 1) & (df['Audit social'] != 0)]

Empty DataFrame
Columns: [code_mission, type_mission, exercice, code_client, total_assistant, manager, code_article, libelle_article, temps, valorisation_temps, site, facturation, entrée_clt, sortie_clt, secteur, forme, ags, cac, cap, cat, com, ge0, ge1, ge2, ge3, gec, gej, gep, ges, gj0, gj1, gj2, gj3, gj4, gj5, gj6, gj7, gs0, gs1, gs2, gs3, gs4, jur, pol, sap, sop, tb, accompagnement dans le suivi de vos travaux, accompagnement fonction daf, assistance au controle fiscal, assistance au controle urssaf, assistance commissaire aux comptes, assistance informatique client, attestations particulires, audit  diagnostic, audit bulletins de paie etou charges sociales, audit des comptes  rdaction des rapports, audit interne dossiers, audit social, autres declarations fiscales, autres missions commissariat, autres prestations sociales, autres travaux en social, avertissement, bilan retraite, budget previsionnel, cerfa apprentissage, cerfa contrat de professionalisation, charte teletravail, circularisation, collecte des lments  inventaire, comptes annuels  et declarations fiscales, contrle des comptes, contrle interne, controle et revision, convention abandon de compte courant, convention de mutation, courrier reponse salarie, cration  migration silae, cration de dossier, decision unilaterale employeur, declaration annuelle dads, declaration arret de travail, declarations mensuelles  trimestrielles, declarations tva, demande chomage partiel, demande renseignements impots, denonciation dusage, diagnostic rgpd, documentation, dossier de participation au regime chomage, dsn evenementielle, etablissement de dads, etablissement de la dsn mensuelle, evaluation, facturation et gestion des comptes clients, fin de periode dessai, forfait demarrage , forfait exo charges patronales et aide au paiement , forfait gestion activite partielledi, ...]
Index: []

[0 rows x 179 columns]

In [95]:
#Création famille Formalités embauche
df['Formalités embauche'] =  df['formalites dembauche'] + df['cerfa apprentissage'] + df['cerfa contrat de professionalisation'] + df['mouvement de personnel entree'] + df['mouvement de personnel entree apprenti'] + df['mouvement de personnel entree cddmoisextrasaisonnier'] + df['contrat de travail et/ou avenant']

In [96]:
df[(df['Formalités embauche'] != 1) & (df['Formalités embauche'] != 0)]

code_mission type_mission exercice code_client  total_assistant  \
4451   AGES2017010108900          GES     2017        5649              5.0   
4453   AGES2017010109100          GES     2017        6474              3.0   
4456   AGES2017010112900          GES     2017        4951              4.0   
4457   AGES2017010113000          GES     2017       79474              6.0   
4458   AGES2017010113200          GES     2017        7642              5.0   
...                  ...          ...      ...         ...              ...   
35473  ASOP2020013221800          SOP     2020       A0647             12.0   
35482  ASOP2020013228000          SOP     2020        4154              2.0   
35486  ASOP2020013247700          SOP     2020       A2088              1.0   
35507  ASOP2020013403400          SOP     2020       A1891              7.0   
35528  ASOP2020013562300          SOP     2020       A2973              4.0   

                     manager  \
4451                     SLE   
4453                     STL   
4456   _OLP NE PLUS UTILISER   
4457                     GSO   
4458                     JOP   
...                      ...   
35473                    THB   
35482                    JOP   
35486                    VAL   
35507                    JOD   
35528                    CBA   

                                            code_article  \
4451                                      ['025', '023']   
4453                                      ['025', '023']   
4456                        ['023', '025', '029', '024']   
4457               ['023', '025', '045', '029', 'F0220']   
4458                               ['025', '023', '024']   
...                                                  ...   
35473  ['F0201', 'F0205', 'F02082', 'F0200', '029', '...   
35482  ['030', '04950', '029', 'F0207', 'F0205', 'F02...   
35486  ['F0200', 'F02081', 'F02052', '020', '029', '0...   
35507  ['030', '029', '020', '0291', '001', 'F0207', ...   
35528  ['001', '030', '029', 'F0207', 'F0205', 'F0208...   

                                         libelle_article   temps  \
4451           contrats de travail, formalites dembauche   12.50   
4453           contrats de travail, formalites dembauche    2.75   
4456   formalites dembauche, contrats de travail, aut...   30.00   
4457   formalites dembauche, contrats de travail, ass...   16.25   
4458   contrats de travail, formalites dembauche, for...   26.75   
...                                                  ...     ...   
35473  bulletin de paie simulation, mouvement de pers...  282.75   
35482  cration  migration silae, travaux exceptionnel...   64.75   
35486  bulletin de paie, forfait mensuel   salaries ,...   49.00   
35507  cration  migration silae, autres travaux en so...   77.50   
35528  renseignements client, cration  migration sila...   28.75   

       valorisation_temps  ... Coffre fort numérique  \
4451               662.25  ...                     0   
4453               138.75  ...                     0   
4456              1820.75  ...                     0   
4457              1181.50  ...                     0   
4458              1685.50  ...                     0   
...                   ...  ...                   ...   
35473            11400.42  ...                     0   
35482             3274.02  ...                     0   
35486             2280.51  ...                     0   
35507             4260.99  ...                     0   
35528             1771.41  ...                     0   

       contrat de travail et/ou avenant  Contrôle des charges  \
4451                                  1                     0   
4453                                  1                     0   
4456                                  1                     0   
4457                                  1                     0   
4458                                  1                     0   
...                                 ...                   ...   
35473       

In [97]:
#Remplacer les autres valeurs par 1 ou 0
df["Formalités embauche"] = df['Formalités embauche'].map({0:0, 1:1, 2:1, 3:1, 4:1, 5:1})

In [98]:
#Supprimer les colonnes agrégées dans la famille d'activité
df.drop(['formalites dembauche', 'cerfa apprentissage', 'cerfa contrat de professionalisation', 'mouvement de personnel entree', 'mouvement de personnel entree apprenti', 'mouvement de personnel entree cddmoisextrasaisonnier', 'contrat de travail et/ou avenant'], axis=1, inplace=True)

In [99]:
#Création famille Sous-traitance
df['Sous-traitance'] =  df['saisie lignes importes'] + df['saisie primo bpo'] + df['hbergement pices comptables'] 

In [100]:
df[(df['Sous-traitance'] != 1) & (df['Sous-traitance'] != 0)]

code_mission type_mission exercice code_client  total_assistant  \
40     ACOM2017008059000          COM     2017       40082              4.0   
279    ACOM2017008223100          COM     2017        6129              3.0   
312    ACOM2017008227400          COM     2017       40032              5.0   
316    ACOM2017008227900          COM     2017       40066              7.0   
465    ACOM2017008247900          COM     2017        5954              3.0   
467    ACOM2017008248100          COM     2017        5956              7.0   
473    ACOM2017008249100          COM     2017        6139              5.0   
1196   ACOM2017008367900          COM     2017        6944              3.0   
1227   ACOM2017008488000          COM     2017        6988              4.0   
1332   ACOM2017008594000          COM     2017        7135              7.0   
1333   ACOM2017008594300          COM     2017        7136              7.0   
1388   ACOM2017008687700          COM     2017        7498              4.0   
1481   ACOM2017008958700          COM     2017        7508              4.0   
1914   ACOM2017009196600          COM     2017        5036              7.0   
1939   ACOM2017009199800          COM     2017        5159              5.0   
1971   ACOM2017009203700          COM     2017        5425              5.0   
2007   ACOM2017009207800          COM     2017        5563              5.0   
2142   ACOM2017009222900          COM     2017        6123              4.0   
2143   ACOM2017009223000          COM     2017        6124              4.0   
2159   ACOM2017009225200          COM     2017        6209              5.0   
2163   ACOM2017009225600          COM     2017        6248              2.0   
2230   ACOM2017009234200          COM     2017        6573              5.0   
2231   ACOM2017009234300          COM     2017        6574              5.0   
2278   ACOM2017009240500          COM     2017        6757              7.0   
2481   ACOM2017009423400          COM     2017        7910              8.0   
2533   ACOM2017009555500          COM     2017        7986              6.0   
2539   ACOM2017009560700          COM     2017        7968              6.0   
2960   ACOM2017010140600          COM     2017       A0520              5.0   
3407   ACOM2017010581200          COM     2017       A0651              9.0   
7900   ACOM2018008594100          COM     2018        7135              7.0   
7901   ACOM2018008594400          COM     2018        7136              7.0   
7926   ACOM2018008687800          COM     2018        7498              4.0   
7942   ACOM2018008713200          COM     2018       40082              4.0   
8168   ACOM2018008769200          COM     2018        6129              5.0   
8259   ACOM2018008958800          COM     2018        7508              5.0   
9147   ACOM2018009246700          COM     2018        6944              6.0   
9163   ACOM2018009248800          COM     2018        6988              4.0   
9627   ACOM2018009329100          COM     2018        5425              6.0   
9662   ACOM2018009333200          COM     2018        5563              5.0   
9779   ACOM2018009348200          COM     2018        6123              7.0   
9780   ACOM2018009348300          COM     2018        6124              6.0   
9856   ACOM2018009359400          COM     2018        6573              5.0   
9857   ACOM2018009359500          COM     2018        6574              6.0   
9900   ACOM2018009365700          COM     2018        6757              5.0   
10083  ACOM2018009423500          COM     2018        7910              6.0   
10129  ACOM2018009560800          COM     2018        7968              6.0   
10450  ACOM2018010140700          COM     2018       A0520              8.0   
10920  ACOM2018010581300          COM     2018       A0651              6.0   
11119  ACOM2018011495900          COM     2018       A0922              4.0   
11233  ACOM2018011816300          COM     2018       A1063    

In [101]:
#Remplacer les autres valeurs par 1 ou 0
df['Sous-traitance'] = df['Sous-traitance'].map({0:0, 1:1, 2:1, 3:1})

In [102]:
#Supprimer les colonnes agrégées dans la famille d'activité
df.drop(['saisie lignes importes', 'saisie primo bpo', 'hbergement pices comptables'], axis=1, inplace=True)

In [103]:
#Création famille Tableau de bord
df['Tableau de bord'] =  df['tb'] + df['tableau de bord']

In [104]:
df[(df['Tableau de bord'] != 1) & (df['Tableau de bord'] != 0)]

code_mission type_mission exercice code_client  total_assistant  \
16855  ATB 2018010659800           TB     2018       A0576              2.0   
16856  ATB 2018010778600           TB     2018       A0648              2.0   
16857  ATB 2018010797200           TB     2018       A0523              4.0   
16858  ATB 2018012107200           TB     2018        7302              1.0   
16859  ATB 2018012107700           TB     2018        7315              1.0   
25689  ATB 2019011968300           TB     2019       A0576              2.0   
25690  ATB 2019011983300           TB     2019       A0648              1.0   
25691  ATB 2019012100400           TB     2019       A0016              3.0   
25692  ATB 2019012107300           TB     2019        7302              2.0   
25693  ATB 2019012107800           TB     2019        7315              2.0   
25694  ATB 2019012140600           TB     2019       A1290              2.0   
35570  ATB 2020012107400           TB     2020        7302              3.0   
35571  ATB 2020012107900           TB     2020        7315              3.0   
35572  ATB 2020012307000           TB     2020       A1443              1.0   
35573  ATB 2020012323900           TB     2020       A1290              2.0   
35574  ATB 2020012940800           TB     2020       A0987              1.0   
35575  ATB 2020012986700           TB     2020        7245              1.0   
35576  ATB 2020012989600           TB     2020       A1205              1.0   
35577  ATB 2020013266500           TB     2020       A1488              2.0   
35578  ATB 2020013408500           TB     2020        7942              1.0   
35579  ATB 2020013461000           TB     2020       A0863              1.0   
35580  ATB 2020013462900           TB     2020       A2885              1.0   
35581  ATB 2020013967000           TB     2020       A1254              1.0   
35582  ATB 2020014179000           TB     2020        5101              1.0   

      manager           code_article  \
16855     JOP                ['043']   
16856     JOP                ['043']   
16857     FRL  ['049', '011', '043']   
16858     NAB             ['F07000']   
16859     NAB             ['F07000']   
25689     JOP                ['043']   
25690     JOP                ['043']   
25691     VAL      ['F07000', '043']   
25692     NAB      ['F07000', '043']   
25693     NAB      ['F07000', '043']   
25694     JOP                ['043']   
35570     NAB      ['043', 'F07000']   
35571     NAB      ['043', 'F07000']   
35572     GSO                ['043']   
35573     JOP                ['043']   
35574     SLE                ['043']   
35575     NAB                ['043']   
35576     GSO                ['043']   
35577     JOP                ['043']   
35578     GSO                ['043']   
35579     GSO                ['043']   
35580     GSO                ['043']   
35581     JOP                ['043']   
35582     GSO                ['043']   

                                         libelle_article  temps  \
16855                                    tableau de bord   7.25   
16856                                    tableau de bord   9.50   
16857  autres travaux exceptionnels, controle et revi...  39.75   
16858                                    tableau de bord   4.00   
16859                                    tableau de bord   4.00   
25689                                    tableau de bord  13.75   
25690                                    tableau de bord   2.00   
25691                   tableau de bord, tableau de bord  18.50   
25692                   tableau de bord, tableau de bord   1.50   
25693                   tableau de bord, tableau de bord   1.50   
25694                                    tableau de bord   7.50   
35570                   tableau de bord, tableau de bord   7.45   
35571                   tableau de bord, tableau de bord   7.20   
35572                                    tableau de bord   2.25   
35573                         

In [105]:
#Remplacer les autres valeurs par 1 ou 0
df['Tableau de bord'] = df['Tableau de bord'].map({0:0, 1:1, 2:1, 3:1})

In [106]:
#Supprimer les colonnes agrégées dans la famille d'activité
df.drop(['tb', 'tableau de bord'], axis=1, inplace=True)

In [107]:
#Supprimer les colonnes qui correspondent aux activités non facturables
df.drop(['non facturable commissariat', 'non facturable comptabilite', 'non facturable juridique', 'non facturable social'], axis=1, inplace=True)

In [108]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35583 entries, 0 to 35582
Columns: 166 entries, code_mission to Tableau de bord
dtypes: float64(4), int64(152), object(10)
memory usage: 45.1+ MB


## Bag of words sur les formes juridiques

In [109]:
text_data_forme = np.array(df['forme'].values)
text_data_forme

array(['SAS', 'EURL', 'SARL', ..., 'SAS', 'SAS', 'EURL'], dtype=object)

In [110]:
# Create the bag of words feature matrix
count = CountVectorizer()
bow_forme = count.fit_transform(text_data_forme)

# Show feature matrix
bow_forme.toarray()

array([[0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       ...,
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0],
       [0, 0, 0, ..., 0, 0, 0]], dtype=int64)

In [111]:
feature_names = count.get_feature_names()

In [112]:
df_formes = pd.DataFrame(bow_forme.toarray(), columns=feature_names)
df_formes

ass  coop  divers  earl  eirl  eurl  gaec  gfa  gfo  gfr  ...  scf  \
0        0     0       0     0     0     0     0    0    0    0  ...    0   
1        0     0       0     0     0     1     0    0    0    0  ...    0   
2        0     0       0     0     0     0     0    0    0    0  ...    0   
3        0     0       0     0     0     0     0    0    0    0  ...    0   
4        0     0       0     0     0     0     0    0    0    0  ...    0   
...    ...   ...     ...   ...   ...   ...   ...  ...  ...  ...  ...  ...   
35578    0     0       0     0     0     1     0    0    0    0  ...    0   
35579    0     0       0     0     0     0     0    0    0    0  ...    0   
35580    0     0       0     0     0     0     0    0    0    0  ...    0   
35581    0     0       0     0     0     0     0    0    0    0  ...    0   
35582    0     0       0     0     0     1     0    0    0    0  ...    0   

       sci  scm  scp  sdf  selarl  selas  sem  sep  snc  
0        0    0    0    0       0      0    0    0    0  
1        0    0    0    0       0      0    0    0    0  
2        0    0    0    0       0      0    0    0    0  
3        0    0    0    0       0      0    0    0    0  
4        0    0    0    0       0      0    0    0    0  
...    ...  ...  ...  ...     ...    ...  ...  ...  ...  
35578    0    0    0    0       0      0    0    0    0  
35579    0    0    0    0       0      0    0    0    0  
35580    0    0    0    0       0      0    0    0    0  
35581    0    0    0    0       0      0    0    0    0  
35582    0    0    0    0       0      0    0    0    0  

[35583 rows x 31 columns]

In [113]:
df = pd.concat([df, df_formes], axis=1)

In [114]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35583 entries, 0 to 35582
Columns: 197 entries, code_mission to snc
dtypes: float64(4), int64(183), object(10)
memory usage: 53.5+ MB


In [115]:
df.forme.value_counts()

SARL           9548
IND            6021
SCI            5728
EURL           4451
SAS            3613
SASU           2019
Particulier     970
ASS             615
SC              548
SELARL          444
SCM             358
EIRL            247
SNC             210
SA              200
Divers          128
EARL            125
SCEA            120
SCP              81
SEP              26
GIE              22
GFA              19
SELAS            16
GFR              15
GAEC             12
GFO              11
INDIV            11
SCF               8
SDF               8
COOP              4
SADIR             3
SEM               2
Name: forme, dtype: int64

In [116]:
df.columns[166:]

Index(['ass', 'coop', 'divers', 'earl', 'eirl', 'eurl', 'gaec', 'gfa', 'gfo',
       'gfr', 'gie', 'ind', 'indiv', 'particulier', 'sa', 'sadir', 'sarl',
       'sas', 'sasu', 'sc', 'scea', 'scf', 'sci', 'scm', 'scp', 'sdf',
       'selarl', 'selas', 'sem', 'sep', 'snc'],
      dtype='object')

## Exportation du dataset en csv pour analyse et machine learning

In [117]:
df.to_csv("./Data/dataset-ML.csv", index=False)